fdbserver/BlobManager.actor.cpp

```diff
@@ -873,38 +840,78 @@ ACTOR Future<Void> blobManager(LocalityData locality, Reference<AsyncVar<ServerD
 		printf("Blob manager taking lock\n");
 	}
 
-	int64_t _epoch = wait(acquireManagerLock(&self));
-	self.epoch = _epoch;
+	self.epoch = epoch;
+
+	// make sure the epoch hasn't gotten stale
+	state Reference<ReadYourWritesTransaction> tr = makeReference<ReadYourWritesTransaction>(self.db);
+
+	loop {
```

Comment:
I decided to `loop` here because there's two cases:
1. `checkManagerLock` throws, in which case we actually rethrow and exit the loop, as expected
2. `commit` throws. This might happen because of a read-write conflict, or for some other reason (maybe?). Regardless, if we try again, and a new manager did enter, then the second time around, `checkManagerLock` will throw and we'll be in case (1).

Reply:
I also noticed that `maybeSplitRange` calls `checkManagerLock` but doesn't do anything different if the BM is old (i.e. `currentEpoch != bmData->epoch`). Should we also be killing the BM in that case?

Reply:
So I think that, as this check is essentially an optimization, since the manager will catch that a new manager was recruited later either by getting sent a halt or getting a later conflict, we don't have to be super strict here. We could just try it once, and if we get some sort or error, we can fail.
That also means we could avoid committing the transaction and just read the lock value, ignoring concurrent transactions.

For the check in maybeSplitRange, it looks like currently the conflict error would get thrown, propagated back up to the actor collection, and it would kill the blob manager with an assert failure.
We probably don't want the assert failure, so we should probably instead just signal the iAmReplaced promise stream if maybeSplitRange gets a conflict

Reply:
Also let's remove nukeBlobWorkerData, as now there can be multiple managers and that was mainly for testing.